### PR TITLE
RDM-4345 - add retry for ES index verification

### PR DESCRIPTION
**Pull request by gbenadikar**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-4345

### Change description ###
Add retries to ES index verification in case ES has been setup in asynchronous mode

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: aat/src/aat/java/uk/gov/hmcts/ccd/definitionstore/tests/functional/elasticsearch/ElasticsearchImportDefinitionTest.java [hunk 1]
```diff
@@ -64,7 +70,29 @@ void shouldImportValidDefinitionMultipleTimes() {
...
 
-        verifyIndexAndFieldMappings(true);
+        verifyIndexAndFieldMappingsWithRetries(true);
+    }
+
+    private void verifyIndexAndFieldMappingsWithRetries(boolean verifyNewFields) {
+        int retry = 1;
+        while (retry <= RETRY_COUNT) {
+            try {
+                verifyIndexAndFieldMappings(verifyNewFields);
```
Comment (smathangi) [inline aat/src/aat/java/uk/gov/hmcts/ccd/definitionstore/tests/functional/elasticsearch/ElasticsearchImportDefinitionTest.java]: Take a look at this test util library. 
https://github.com/awaitility/awaitility/wiki/Usage#example-7---java-8

Comment (gbenadikar) [reply aat/src/aat/java/uk/gov/hmcts/ccd/definitionstore/tests/functional/elasticsearch/ElasticsearchImportDefinitionTest.java]: wonderful sateesh, thanks for that. Good utility.

File: aat/src/aat/java/uk/gov/hmcts/ccd/definitionstore/tests/functional/elasticsearch/ElasticsearchImportDefinitionTest.java [hunk 2]
```diff
@@ -64,13 +69,29 @@ void shouldImportValidDefinitionMultipleTimes() {
...
 
-        verifyIndexAndFieldMappings(true);
+        verifyIndexAndFieldMappingsWithRetries(true);
     }
 
-    private void verifyIndexAndFieldMappings(boolean verifyNewFields) {
-        ValidatableResponse response = getIndexInformation(CASE_INDEX_ALIAS);
-        verifyIndexAndAlias(response);
-        verifyCaseDataFields(response, verifyNewFields);
+    private void verifyIndexAndFieldMappingsWithRetries(boolean verifyNewFields) {
```
Comment (mario-paniccia) [inline aat/src/aat/java/uk/gov/hmcts/ccd/definitionstore/tests/functional/elasticsearch/ElasticsearchImportDefinitionTest.java]: you could call make this method more generic by calling it `withRetries` and passing a Java function as argument and a String description:
`withRetries(() -> verifyIndexAndFieldMappings(verifyNewFields), "ES index verification")

Comment (gbenadikar) [reply aat/src/aat/java/uk/gov/hmcts/ccd/definitionstore/tests/functional/elasticsearch/ElasticsearchImportDefinitionTest.java]: added withRetries in a utility class

File: aat/src/aat/java/uk/gov/hmcts/ccd/definitionstore/tests/functional/elasticsearch/ElasticsearchImportDefinitionTest.java [hunk 3]
```diff
@@ -64,13 +70,19 @@ void shouldImportValidDefinitionMultipleTimes() {
...
-        ValidatableResponse response = getIndexInformation(CASE_INDEX_ALIAS);
-        verifyIndexAndAlias(response);
-        verifyCaseDataFields(response, verifyNewFields);
+    private boolean verifyIndexAndFieldMappings(boolean verifyNewFields) {
+        try {
+            ValidatableResponse response = getIndexInformation(CASE_INDEX_ALIAS);
+            verifyIndexAndAlias(response);
+            verifyCaseDataFields(response, verifyNewFields);
+        } catch (AssertionError e) {
+            log.warn("Retrying Elasticsearch index api due to error: {}", e.getMessage());
```
Comment (mario-paniccia) [inline aat/src/aat/java/uk/gov/hmcts/ccd/definitionstore/tests/functional/elasticsearch/ElasticsearchImportDefinitionTest.java]: shall we change this to INFO?

File: aat/src/aat/java/uk/gov/hmcts/ccd/definitionstore/tests/util/RetryUtility.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package uk.gov.hmcts.ccd.definitionstore.tests.util;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+import static org.awaitility.Awaitility.with;
+
+public class RetryUtility {
```
Comment (mario-paniccia) [inline aat/src/aat/java/uk/gov/hmcts/ccd/definitionstore/tests/util/RetryUtility.java]: shall we call this TestUtils so we can add more utility methods in the future?

Comment (gbenadikar) [reply aat/src/aat/java/uk/gov/hmcts/ccd/definitionstore/tests/util/RetryUtility.java]: done

File: aat/src/aat/java/uk/gov/hmcts/ccd/definitionstore/tests/functional/elasticsearch/ElasticsearchImportDefinitionTest.java (hunk 3 above)
Comment (gbenadikar) [reply aat/src/aat/java/uk/gov/hmcts/ccd/definitionstore/tests/functional/elasticsearch/ElasticsearchImportDefinitionTest.java]: done

**Pull request merged**